analyzers/tests/SonarAnalyzer.UnitTest/Common/MetricsTest.cs

```diff
@@ -563,17 +586,16 @@ End If
 
         private static MetricsBase MetricsFor(AnalyzerLanguage language, string text)
         {
-            if (language != AnalyzerLanguage.CSharp &&
-                language != AnalyzerLanguage.VisualBasic)
+            if (language != AnalyzerLanguage.CSharp && language != AnalyzerLanguage.VisualBasic)
             {
                 throw new ArgumentException("Supplied language is not C# neither VB.Net", nameof(language));
             }
 
-            var (syntaxTree, semanticModel) = TestHelper.Compile(text);
+            var (syntaxTree, semanticModel) = TestHelper.Compile(text, language == AnalyzerLanguage.CSharp);
 
             return language == AnalyzerLanguage.CSharp
                 ? new Metrics.CSharp.CSharpMetrics(syntaxTree, semanticModel)
-                : new Metrics.VisualBasic.VisualBasicMetrics(VisualBasicSyntaxTree.ParseText(text), semanticModel);
+                : new Metrics.VisualBasic.VisualBasicMetrics(syntaxTree, semanticModel);
```

Comment:
All VB test cases were compiled as C# code and then we computed metrics totally wrong